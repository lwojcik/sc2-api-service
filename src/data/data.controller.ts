import { Controller, Get, Param, Query } from '@nestjs/common';
import { ApiOperation, ApiTags } from '@nestjs/swagger';
import { LoggerService } from '../logger/logger.service';
import { DataService } from './data.service';

@ApiTags('data')
@Controller('data')
export class DataController {
  constructor(
    private readonly dataService: DataService,
    private readonly logger: LoggerService
  ) {
    this.logger.setLoggedClass(DataController.name);
  }

  @Get('/league/:seasonId/:queueId/:teamType/:leagueId')
  @ApiOperation({
    summary: 'Get data for the specified season, queue, team, and league.',
  })
  // @ApiOkResponse({
  //   description: ApiResponse.ok,
  //   type: StatusResponse,
  // })
  // @ApiBadRequestResponse({
  //   description: ApiResponse.badRequest,
  // })
  // @ApiUnauthorizedResponse({
  //   description: ApiResponse.unauthorized,
  //   type: UnauthorizedError,
  // })
  // @ApiTooManyRequestsResponse({
  //   description: ApiResponse.tooManyRequests,
  //   type: TooManyRequestsError,
  // })
  getLeague(
    @Param('seasonId') seasonId: number,
    @Param('queueId') queueId: number,
    @Param('teamType') teamType: number,
    @Param('leagueId') leagueId: number,
    @Query('refresh') refresh?: boolean
  ) {
    this.logger.setLoggedMethod(this.getLeague.name);
    this.logger.debug();

    return this.dataService.getLeague(
      {
        seasonId,
        queueId,
        teamType,
        leagueId,
      },
      refresh
    );
  }
}
