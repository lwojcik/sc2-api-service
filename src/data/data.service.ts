import { Injectable } from '@nestjs/common';
import { LoggerService } from '../logger/logger.service';
import { GetLeagueDto } from './dto/get-league.dto';

@Injectable()
export class DataService {
  constructor(private readonly logger: LoggerService) {
    this.logger.setLoggedClass(DataService.name);
  }

  getLeague(getLeagueDto: GetLeagueDto) {
    this.logger.setLoggedMethod(this.getLeague.name);
    this.logger.debug();

    return getLeagueDto;
  }
}
