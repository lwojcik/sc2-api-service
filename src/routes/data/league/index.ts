import { FastifyPlugin } from 'fastify';
import fp from 'fastify-plugin';
import schema from './schema';
import { LeagueObject } from '../../../@types/fastify';

interface RouteQueryString {
  refresh?: boolean;
}

const route: FastifyPlugin = (server, {}, next) => {
  server.route<{
    Querystring: RouteQueryString,
  }>({
    schema,
    url: '/data/league/:seasonId/:queueId/:teamType/:leagueId',
    method: 'GET',
    handler: async (request, reply) => {
      const { refresh } = request.query;
      const data = await server.sc2api.getLeague(
        request.params as LeagueObject,
        refresh,
      );
      reply.code(data.status).send(data);
    },
  });
  next();
};

export default fp(route);
