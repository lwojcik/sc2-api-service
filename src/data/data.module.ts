import { Module } from '@nestjs/common';
import { DataBrokerModule } from '../databroker/databroker.module';
import { LoggerModule } from '../logger/logger.module';
import { DataController } from './data.controller';
import { DataService } from './data.service';

@Module({
  imports: [DataBrokerModule, LoggerModule],
  controllers: [DataController],
  providers: [DataService],
})
export class DataModule {}
