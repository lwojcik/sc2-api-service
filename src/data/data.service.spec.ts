import { Test, TestingModule } from '@nestjs/testing';
import { ModuleMocker } from 'jest-mock';
import { DataService } from './data.service';

const moduleMocker = new ModuleMocker(global);

describe('DataService', () => {
  let service: DataService;

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      providers: [DataService],
    })
      .useMocker((token) => {
        if (typeof token === 'function') {
          const mockMetadata = moduleMocker.getMetadata(token);
          const Mock = moduleMocker.generateFromMetadata(mockMetadata);
          return new Mock();
        }
        return jest.fn();
      })
      .compile();

    service = module.get<DataService>(DataService);
  });

  it('should be defined', () => {
    expect(service).toBeDefined();
  });
});
