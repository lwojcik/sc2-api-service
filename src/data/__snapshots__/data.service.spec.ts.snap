// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataService should get league data 1`] = `
{
  "data": {
    "prop": "sample_league_data_from_mock",
  },
  "status": 200,
}
`;
